core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/SparkRunner.scala

```diff
@@ -194,6 +197,141 @@ object SparkRunner extends Runner[CR] with PatchProvider[SpecPatch] {
+    updatedDriver = podsConfig.pods
+      .get(DriverPod)
+      .orElse(podsConfig.pods.get(PodsConfig.CloudflowPodName))
+      .flatMap { podConfig =>
+        podConfig.containers.get(PodsConfig.CloudflowContainerName).flatMap { containerConfig =>
+          containerConfig.resources.map { resources =>
+            updatedDriver.copy(
+              cores = toIntCores(resources.requests.get(Resource.cpu)).orElse(coresOpt),
+              memory = resources.requests.get(Resource.memory).map(_.toString).orElse(updatedDriver.memory),
+              coreLimit = resources.limits.get(Resource.cpu).map(_.toString).orElse(updatedDriver.coreLimit)
+            )
+          }
+        }
+      }
+      .getOrElse(updatedDriver)
+    log.info(s"""
+    Streamlet ${deployment.streamletName} - resources for driver pod:
+      cores:          ${updatedDriver.cores}
+      memory:         ${updatedDriver.memory}
+      coreLimit:      ${updatedDriver.coreLimit}
+      memoryOverhead: ${updatedDriver.memoryOverhead}
+    """)
+    updatedDriver
+  }
+
+  private def addExecutorResourceRequirements(executor: Executor, podsConfig: PodsConfig, deployment: StreamletDeployment)(
+      implicit ctx: DeploymentContext
+  ): Executor = {
+    var updatedExecutor = executor
+    import ctx.sparkRunnerSettings._
+
+    // this logic means you can only set cores to an int. and it blows up for decimals
+    // From Spark Operator (which is not supported yet):
+    //
+    //   CoreRequest is the physical CPU core request for the executor.
+    //   Maps to `spark.kubernetes.executor.request.cores` that is available since Spark 2.4.
+    //
+    val coresOpt = toIntCores(executorSettings.cores)
+
+    updatedExecutor = updatedExecutor.copy(
+      cores = coresOpt,
+      memory = executorSettings.memory.map(_.value),
+      coreLimit = executorSettings.coreLimit.map(_.value),
+      memoryOverhead = executorSettings.memoryOverhead.map(_.value)
+    )
+    // you can set the "executor" pod or just "pod", which means it will be used for both executor and executor (as fallback).
+    updatedExecutor = podsConfig.pods
+      .get(ExecutorPod)
+      .orElse(podsConfig.pods.get(PodsConfig.CloudflowPodName))
+      .flatMap { podConfig =>
+        podConfig.containers.get(PodsConfig.CloudflowContainerName).flatMap { containerConfig =>
+          containerConfig.resources.map { resources =>
+            updatedExecutor.copy(
+              cores = toIntCores(resources.requests.get(Resource.cpu)).orElse(coresOpt),
+              coreRequest = resources.requests.get(Resource.cpu).map(_.value),
+              memory = resources.requests.get(Resource.memory).map(_.toString).orElse(updatedExecutor.memory),
+              coreLimit = resources.limits.get(Resource.cpu).map(_.toString).orElse(updatedExecutor.coreLimit)
+            )
+          }
+        }
+      }
```

Comment:
Same question for the setting precedence for the executor.

Reply:
it's podConfig resources or else executorSettings, where executorSettings are the defaults.